### PR TITLE
Configure SNMPv3 for Mellanox IB switch

### PR DESCRIPTION
**Pull request by cxhong**
Issue #1912 ,   Add support to configure SNMPv3 for Mellanox IB switch.

### CODE REVIEW

**Comment by cxhong** [comment]
@hu-weihua , can u review it again?  thanks

**Comment by hu-weihua** [comment]
@cxhong , thanks for your change. I don't have more comments.
@zet809 , do you have any comments about switch configuration part?

File: xCAT-server/share/xcat/scripts/configMellanox [hunk 1]
```diff
@@ -267,9 +267,12 @@ sub config_snmp {
         }
 
         $cmd = `xdsh $switch -l $user --devicetype IBSwitch::Mellanox "enable;configure terminal;snmp-server user $snmp_user v3 enable;snmp-server user $snmp_user v3 auth $snmp_auth $snmp_passwd;no snmp-server user $snmp_user v3 require-privacy;configuration write;exit;exit" `;
-
-        $cmd = "chdef $switch status=snmp_configed snmpversion=3 snmpauth=$snmp_auth snmpusername=$snmp_user snmppassword=$snmp_passwd";
-        $rc= xCAT::Utils->runcmd($cmd, 0);
+        if ($::RUNCMD_RC != 0) {
```
Comment (hu-weihua) [inline xCAT-server/share/xcat/scripts/configMellanox]: @cxhong , thanks for your update.
In my memory, ``$::RUNCMD_RC`` is a global attribute defined by xCAT,  so just when we use function ``xCAT::Utils->runcmd`` to trigger a shell command, the return value of this shell command will be saved in ``$::RUNCMD_RC``. But when we use `` ` ` `` to trigger a shell command, just like you did in line 269, the return value of shell command won't be saved in  ``$::RUNCMD_RC``, it still is saved in ``$?``. So I don't  think line 270 will work as you expected.

File: xCAT-server/share/xcat/scripts/configMellanox [hunk 2]
```diff
@@ -224,6 +233,55 @@ sub config_hostname {
...
+        if ($::AUTH) {
+            $snmp_auth = $::AUTH;
+        } elsif ($switchhash->{$switch}->[0]->{auth}) {
+            $snmp_auth = $switchhash->{$switch}->[0]->{auth};
+        } else {
+            $snmp_auth = "sha\r";
+        }
+
+        $cmd = `xdsh $switch -l $user --devicetype IBSwitch::Mellanox "enable;configure terminal;snmp-server user $snmp_user v3 enable;snmp-server user $snmp_user v3 auth $snmp_auth $snmp_passwd;no snmp-server user $snmp_user v3 require-privacy;configuration write;exit;exit" `;
+        if ($::RUNCMD_RC != 0) {
```
Comment (hu-weihua) [inline xCAT-server/share/xcat/scripts/configMellanox]: @cxhong , thanks for your update.
In my memory, $::RUNCMD_RC is a global attribute defined by xCAT, so just when we use function xCAT::Utils->runcmd to trigger a shell command, the return value of this shell command will be saved in $::RUNCMD_RC. But when we use ` ` to trigger a shell command, just like you did in line 269, the return value of shell command won't be saved in $::RUNCMD_RC, it still is saved in $?. So I don't think line 270 will work as you expected.

**Comment by cxhong** [comment]
Thanks, @hu-weihua

**Comment by hu-weihua** [comment]
I have reviewed the change and agree to merge. Thanks @cxhong .

**Pull request merged**